pkg/componentreadiness/jiraintegrator/jiraintegrator.go

```diff
@@ -0,0 +1,524 @@
+					scopeView.SampleRelease.RelativeEnd = resolutionDate.Format(time.RFC3339)
+					scopeReport, err := j.getComponentReportForView(scopeView)
+					if err != nil {
+						logger.WithError(err).Error("error getting report for scope check")
+					}
+
+					// Identify tests only appearing in current report, not scope report
+					scopeRegressedTests := map[crtype.RowIdentification]map[crtype.ColumnID]crtype.ReportTestSummary{}
+					for _, row := range scopeReport.Rows {
+						for _, col := range row.Columns {
+							for _, test := range col.RegressedTests {
+								if _, ok := scopeRegressedTests[test.RowIdentification]; !ok {
+									scopeRegressedTests[row.RowIdentification] = map[crtype.ColumnID]crtype.ReportTestSummary{}
+								}
+								columnKeyBytes, err := json.Marshal(test.ColumnIdentification)
+								if err != nil {
+									return err
+								}
+								scopeRegressedTests[test.RowIdentification][crtype.ColumnID(columnKeyBytes)] = test
+							}
+						}
+					}
+					newTests := []crtype.ReportTestSummary{}
+					for _, test := range tests {
+						columnKeyBytes, err := json.Marshal(test.ColumnIdentification)
+						if err != nil {
+							return err
+						}
+						_, ok := scopeRegressedTests[test.RowIdentification]
+						if !ok {
+							newTests = append(newTests, test)
+						} else if _, ok := scopeRegressedTests[test.RowIdentification][crtype.ColumnID(columnKeyBytes)]; !ok {
+							newTests = append(newTests, test)
+						}
+					}
+					if len(newTests) > 0 {
+						// Any tests not covered by scope check is considered new
+						_, err := j.createNewJiraIssueForRegressions(view, component, newTests, &selected)
+						if err != nil {
+							log.WithError(err).Error("error creating jira issue")
+						}
+					} else {
+						// This means scope report contains all tests from current report, verify fix
+						if resolutionDate.Add(fixCheckMinimumDays).Before(view.SampleRelease.End) {
+							fixView := view
+							fixView.TestIDOption.Component = component
+							fixView.SampleRelease.RelativeStart = resolutionDate.Format(time.RFC3339)
+							fixReport, err := j.getComponentReportForView(fixView)
+							if err != nil {
+								logger.WithError(err).Error("error getting report for fix check")
+							}
+							regressedTests := j.getComponentRegressedTestsFromReport(fixReport)
+							if tests, ok := regressedTests[component]; ok {
+								_, err := j.createNewJiraIssueForRegressions(fixView, component, tests, nil)
+								if err != nil {
+									log.WithError(err).Error("error creating jira issue")
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		// test code. Only process one component for testing purposes
+		break
+	}
+	return nil
+}
+
+func (j JiraIntegrator) IntegrateJira() error {
+	log.Infof("Start integrating component readiness regressions with Jira")
+	for _, view := range j.views {
+		err := j.integrateJiraForView(view)
```

Comment:
You'll want to add configuration on the View type for automated jiras, there are examples there already for turning on metrics and regression tracking.